### ISSUE TITLE
Includes on association with scope containing distinct doesn't make include distinct

### ISSUE THREAD

**Issue by sb8244** [rails/rails#26545]
### Steps to reproduce

Sorry that I haven't yet put together a code reproduction of this. I can do that if needed. I'm boiling down from the test case + fix in my company's application.

Create a scoped association where the scope is `joins(:b).distinct`. Use `A.includes(:b).bs` to see that the b model is duplicated (SQL log shows that distinct isn't used). 
### Expected behavior

The SQL included should have distinct in it, and the result shouldn't have duplicates.
### Actual behavior

The SQL doesn't have distinct in it; items are duplicated.
### System configuration

**Rails version**: 4.2.7

**Ruby version**: 2.2.4
### Ideas to Fix

I saw https://github.com/rails/rails/blob/4-2-stable/activerecord/lib/active_record/associations/preloader/association.rb#L146 and decided to see if I could fix it locally. I added 

```
    scope.distinct! preload_values[:distinct] || values[:distinct]
```

and it passes my test cases locally.

**Comment by sb8244** [comment]
I'm not actually sure those steps to reproduce will work for you. I'm working on coming up with a test case for it, but our relationship is complex and so I'm trying to mentally work through how to distill it

**Comment by maclover7** [comment]
👍 please post here when you have a reproduction script.

**Comment by sb8244** [comment]
Here is reproduction repo @maclover7  https://github.com/sb8244/rails-issue-26545

Sorry that it isn't "stand alone". It's a complex problem to recreate and so I wanted to do so in a full Rails environment rather than a minimum viable script.

The fix is also included there. I'm happy to do a PR similar to https://github.com/rails/rails/commit/91e3dab804fffe0b1daebb438091418faa1fa256 but I couldn't get a meaningful test in the rails spec suite. However, I do not think the existing spec is very meaningful.

**Comment by sb8244** [comment]
Also forgive the weird includes. I know that is how the issue happens, so kept true to it.

**Comment by maclover7** [comment]
Hi @sb8244! Thanks for providing the reproduction application (sidenote: converted it to a more-easily-runnable reproduction script [here](https://gist.github.com/maclover7/ee54dacea920a2580335ffe3d09356b8)! I can also verify that the fix in `problem_spec.rb` seems to solve the problem. Can you open up a PR to Rails with your fix, and try and add a regression test? I can lend a hand if needed. :)

**Comment by sb8244** [comment]
I'll give it a shot! Really appreciate your reproduction script. Now I have
a reference for the future.

**Comment by sb8244** [comment]
Thanks for the help @maclover7 , I just submited a PR on #26617

**Comment by kamipo** [comment]
This should be fixed by 2bfa2c02a891d4817832807343c6f1d56e9bae50.
